fieldpath/serialize.go

```diff
@@ -204,34 +205,25 @@ func readIterV1(iter *jsoniter.Iterator) (children *Set, isMember bool) {
 			if children == nil {
 				children = &Set{}
 			}
-			m := &children.Members.members
-			// Since we expect that most of the time these will have been
-			// serialized in the right order, we just verify that and append.
-			appendOK := len(*m) == 0 || (*m)[len(*m)-1].Less(pe)
-			if appendOK {
-				*m = append(*m, pe)
-			} else {
-				children.Members.Insert(pe)
```

Comment:
does append + sort do the same thing as Insert (here) and Descend (below)? 

it looks like Insert _can_ be a no-op if the item already exists in the set:
https://github.com/kubernetes-sigs/structured-merge-diff/blob/3392408e4daa0089f96601a9ce96596f610ec813/fieldpath/element.go#L265-L267

and Descend _can_ return the existing set if the path already exists

https://github.com/kubernetes-sigs/structured-merge-diff/blob/3392408e4daa0089f96601a9ce96596f610ec813/fieldpath/set.go#L496-L498

It looks like switching these to unconditional appends can produce duplicates where that was impossible before?

If so, can you add tests that ensure this isn't changing behavior?